### ISSUE TITLE
Failing HTTP request causes Unhandled Promise Rejection

### ISSUE THREAD

**Issue by matejkramny** [john-doherty/mixpanel-lite#5]
Hello, I found an issue whilst using this library.

When using adblockers some requests to mixpanel may get rejected.
This causes the request to fail, and the promise to get rejected ([here](https://github.com/john-doherty/mixpanel-lite/blob/master/src/mixpanel-lite.js#L526)).

[This code](https://github.com/john-doherty/mixpanel-lite/blob/master/src/mixpanel-lite.js#L248) should have a catch() to handle any thrown errors by the promise.

I've made a [fork](https://github.com/matejkramny/mixpanel-lite) that fixes this although it pretends the request succeeded if errors are thrown.

I haven't looked enough, maybe you can help me out. If requests to mixpanel are permanently blocked, would the code repeatedly try already failed requests (not marked as __complete) every time a new event is created?

**Comment by john-doherty** [comment]
Hi there, good find. Yes, it would continue trying indefinitely which is not ideal... Let me give this some thought and come back to you.

**Comment by john-doherty** [comment]
Fixed in [refactor](https://github.com/john-doherty/mixpanel-lite/commit/81b7bfedaca1d3cc9d2545b32d65386af5cd3dea)

**Comment by john-doherty** [comment]
@matejkramny I've had to update this. If a request fails to Mixpanel, it should stop all requests, otherwise, the events are sent in the wrong order. If requests are blocked via an ad blocker, they should all fail as they're stacked in the order they occurred.

Yes, the code would repeatedly try to send with every new `.track` call. To avoid localStorage bloat, I've set a max number of offline requests to 250, new events added after that will cause the first event to be dropped (fifo style).
